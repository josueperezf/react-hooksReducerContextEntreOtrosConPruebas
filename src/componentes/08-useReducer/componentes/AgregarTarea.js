import React from 'react'
import { useForm } from '../../../hooks/useForm';

export const AgregarTarea = ({handleAgregarTarea}) => {
    const [{descripcion}, handleInputChange, reset] = useForm ({
        descripcion: ''
    });
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (descripcion.trim().length ===0) {
            return;
        }
        const nuevaTarea = {
            id:     new Date().getTime(),
            desc:   descripcion,
            done:   false
        };
        handleAgregarTarea(nuevaTarea);
        reset();
    }
    return (
        <>
            <h4>Agregar Tarea</h4>
            <hr/>
            <form onSubmit={handleSubmit} >
                <input
                    type='text'
                    name='descripcion'
                    value={descripcion}
                    className='form-control'
                    placeholder='Aprender...'
                    autoComplete='off'
                    onChange={ handleInputChange }
                />
                <button
                    className='btn btn-outline-primary mt-1 w-100'
                    type='submit'
                >
                    Agregar
                </button>
            </form>
        </>
    )
}
