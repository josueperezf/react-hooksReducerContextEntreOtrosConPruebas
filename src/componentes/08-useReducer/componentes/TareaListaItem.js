import React from 'react'

export const TareaListaItem = ({tarea, i, handleDelete, handleToggle}) => {
    return (
        <li className='list-group-item'  key={tarea.id} >
            <p className={ `${tarea.done && 'complete'}`}
                onClick={()=>handleToggle(tarea.id)}
            >
                {i+1 }. {tarea.desc}
            </p>
            <button
                className='btn btn-danger'
                onClick={()=>handleDelete(tarea.id)}
            >
                Borrar
            </button>
        </li>
    )
}
