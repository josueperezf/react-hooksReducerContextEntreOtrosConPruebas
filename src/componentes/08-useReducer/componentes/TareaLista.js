import React from 'react'
import { TareaListaItem } from './TareaListaItem';
export const TareaLista = ({tareas, handleDelete, handleToggle}) => {
    return (
        <ul className="list-group list-group-flush">
            {
                tareas.map((tarea, i)=>(
                    <TareaListaItem key={tarea.id}  tarea={tarea} i={i} handleDelete={handleDelete} handleToggle={handleToggle} />
                ))
            }
        </ul>
    )
}
