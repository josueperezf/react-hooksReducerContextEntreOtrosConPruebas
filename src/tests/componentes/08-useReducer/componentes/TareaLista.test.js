import { shallow } from "enzyme";
import { TareaLista } from "../../../../componentes/08-useReducer/componentes/TareaLista";
import { listadoTareasDePrueba } from "../../../fixtures/listadoTareasDePrueba";


describe('Prueba del componente <TareaLista />', ()=>{
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<TareaLista
                                tareas={listadoTareasDePrueba}
                                handleDelete={handleDelete}
                                handleToggle={handleToggle}
                            />);
    test('Debe Mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe dibujar varios <TareaListaItem/> segun el length de arreglo enviado', () => {
        expect(wrapper.find('TareaListaItem').length).toBe(listadoTareasDePrueba.length);
        // verificamos si tiene las funciones, Function es para el tipo de dato
        expect(wrapper.find('TareaListaItem').at(0).prop('handleDelete')).toEqual(expect.any(Function) );
        expect(wrapper.find('TareaListaItem').at(0).prop('handleToggle')).toEqual(expect.any(Function) );
    })
    
    
});