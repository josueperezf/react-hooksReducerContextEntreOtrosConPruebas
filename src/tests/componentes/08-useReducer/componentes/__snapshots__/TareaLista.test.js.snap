// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <TareaLista /> Debe Mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TareaListaItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    tarea={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TareaListaItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    tarea={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
